plugin/pkg/shoot/validator/admission.go

```diff
@@ -236,9 +244,9 @@ func (v *ValidateShoot) Admit(ctx context.Context, a admission.Attributes, _ adm
 		}
 	}
 
-	cloudProfile, err := v.cloudProfileLister.Get(shoot.Spec.CloudProfileName)
+	cloudProfile, err := admissionutils.GetCloudProfile(v.cloudProfileLister, v.namespacedCloudProfileLister, shoot.Spec.CloudProfileName, shoot.Namespace)
 	if err != nil {
-		return apierrors.NewInternalError(fmt.Errorf("could not find referenced cloud profile: %+v", err.Error()))
+		return apierrors.NewInternalError(fmt.Errorf("could not find referenced (private) cloud profile: %+v", err.Error()))
```

Comment:
Why are we not using the new `CloudProfileReference` field?

Reply:
The shoot's API change is only introduced as of this PR but without any functionality. This is intentional as this PR is intended to introduce API changes to allow provider extensions to adapt to the new API. A future PR will define a migration plan from the `cloudProfileName` field to the new `cloudProfile` field. TL;DR: the field will be put to use in a future PR

Reply:
Hm, with this argument it seems odd that you introduce this utility because now one can think that `shoot.Spec.CloudProfileName` can refer to a `NamespacedCloudProfile`. It should be safe and consistent to use `shoot.Spec.CloudProfileName` and `spec.cloudProfile` here.

Reply:
I have noticed that the `spec.CloudProfile` field and corresponding admission plugins / utils are causing confusion. They are subject to major change in future NamespacedCloudProfile PR's and are not needed for this PR. Therefore, I removed them from the PR. Sorry for the inconvenience.